crates/sui-core/src/authority_server.rs

```diff
@@ -486,71 +558,86 @@ impl ValidatorService {
             // For shared objects this will wait until either timeout or we have heard back from consensus.
             // For owned objects this will return without waiting for certificate to be sequenced
             // First do quick dirty non-async check
-            if !epoch_store.is_tx_cert_consensus_message_processed(&certificate)? {
+            if !epoch_store.is_all_tx_certs_consensus_message_processed(&verified_certificates)? {
```

Comment:
iiuc, this means that if a single transaction in the bundle is unprocessed, we will re-submit the entire bundle. That's okay from the protocol perspective, but is that what is desired for the submitter of the soft bundle? Or would they rather fail in this case, since it is known their bundle will not be sequenced in the order they expect?

Reply:
The intention here is to wait for all txs to be sequenced before proceeding.  In a Soft Bundle context, the check was done previously (by using `self.state.is_tx_already_executed()`).

However this check cannot be enforced as another thread or another validator may be submitting a subset of the bundle and/or the entire bundle.

This falls into the case where Soft Bundle cannot guarantee specified ordering.